tests/phpunit/integration/Modules/AdSense/Tag_GuardTest.php

```diff
@@ -22,44 +22,48 @@
  */
 class Tag_GuardTest extends TestCase {
 
-	public function test_can_activate() {
-		$settings = new Settings( new Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ) );
-		$guard    = new Tag_Guard( $settings );
+	protected function get_guard( array $options = array() ) {
+		static $guard = null;
+
+		if ( is_null( $guard ) ) {
+			$settings = new Settings( new Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ) );
+			$guard    = new Tag_Guard( $settings );
+		}
 
 		update_option(
 			Settings::OPTION,
-			array(
-				'clientID'   => 'test-client-id',
-				'useSnippet' => true,
+			array_merge(
+				array(
+					'clientID'   => 'test-client-id',
+					'useSnippet' => true,
+				),
+				$options
```

Comment:
Let's do this in a `setUp` method instead rather than as a side-effect of this method.

Then if we stored the `$settings` as an instance property, it would be available everywhere it's needed, both as a dependency of the `Tag_Guard` class, and for setting/merging settings as part of test set up. Then the `get_guard` method isn't really necessary anymore, and there would be no duplicate lines instantiating common dependencies or settings.

Reply:
Updated.